UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs

```diff
@@ -78,6 +91,44 @@ public ExchangeSetServiceResponse CreateExchangeSetForPostProductIdentifier(stri
             return null;
         }
 
+        public ExchangeSetServiceResponse CreateExchangeSetForPostProductVersion(List<ProductVersionRequest> productVersionsRequest)
+        {
+            CreateBatchRequest batchRequest;
+
+            foreach (ProductVersionRequest? item in productVersionsRequest)
+            {
+                if (item.ProductName.Contains("GB800001") ||
+                    item.ProductName.Contains("GA800001") ||
+                    item.ProductName.Contains("GC800001"))
+                {
+                    batchRequest = CreateBatchRequestModelForAIO(false);
+                }
+                else
+                {
+                    batchRequest = CreateBatchRequestModel(true);
+                }
+
+                BatchResponse createBatchResponse = _fssService.CreateBatch(batchRequest.Attributes, _homeDirectoryPath);
+                string productVersion = $"productVersion-{item.ProductName}-{item.EditionNumber}-{item.UpdateNumber}";
+
+                if (!string.IsNullOrEmpty(createBatchResponse.BatchId.ToString()))
+                {
+                    string path = Path.Combine(Environment.CurrentDirectory, @"Data", createBatchResponse.BatchId.ToString());
+                    foreach (string fileName in Directory.GetFiles(path))
+                    {
+                        FileInfo file = new(fileName);
```

Comment:
Generate a FileInfo object from the fileName, but then only use it to access the fileName (line 120). Perhaps easier to just use the fileName directy

Reply:
The filename used in the Fileinfo object is in fact the file path. Therefore renamed the variable.